### PR TITLE
Feature/add print model

### PR DESCRIPTION
**Pull request by jgillies**
I thought it would be handy to be able to print models to STDOUT, rather than opening them in a text editor. My specific use-case is to pipe the STDOUT into a new Vim buffer to avoid having to leave Vim to view the models.

### CODE REVIEW

**Comment by mikekaminsky** [comment]
@jgillies neat! how are you actually doing this with vim? (also cc @clrcrl)

File: core/open.py [hunk 1]
```diff
@@ -95,15 +94,20 @@ def run(self):
...
             )
 
-        result = subprocess.call(
-            " ".join([self.open_command, file_to_open]), shell=True
-        )
-        if result == 0:
-            print("Opened " + file_to_open)
+        if self.args.print_model:
+            with open(file_to_open, "r") as f:
+                print(f.read())
```
Comment (mikekaminsky) [inline core/open.py]: @jgillies Any ideas for a way to test this in our integration test suite?

Comment (jgillies) [reply core/open.py]: @mikekaminsky Unfortunately, no. Python isn't really my bag, so I was kind of winging it here.

Comment (mikekaminsky) [reply core/open.py]: Haha okay i'll take a look either this afternoon or tomorrow and see if I can come up with something :)

**Comment by jgillies** [comment]
@mikekaminsky Here's the line from my Vim config: `nnoremap <leader>go :!dbt-helper open <cword> --print<CR>`. That _should_ open the file for the model under the cursor.

**Comment by mikekaminsky** [comment]
@jgillies extremely slick! I'm going to try to figure out a way to add a test for this but if I can't figure that out in a few days I'm happy to ship it.

**Comment by jgillies** [comment]
@mikekaminsky Great! This is my first-ever contribution to an open source project!

**Comment by mikekaminsky** [comment]
@jgillies now that I'm thinking about this ... I'm not sure this is doing what you want. I think what you want is to have dbt-helper give you the _path_ to the file that you want to open so that you can open that in vim. Right now, I think vim is going to open a buffer and put the result of STDOUT (which is the _code_ for the model, but not the file where that code lives).

Does this make sense?

**Comment by clrcrl** [comment]
I have some thoughts about the integration tests here, and if you _do_ just want the path, then these thoughts are still valid.

I think the better version of the integration tests should _test whether the correct path is returned_. To do this, I think the `OpenTask.run` method should return both the `result` and `file_to_open`, i.e. [this line](https://github.com/mikekaminsky/dbt-helper/blob/dd657844a5d4680bc2c7fbf4fd2af862891e35c3/core/open.py#L113) should have:
```
return result, file_to_open
```

I thought about changing this a while back, but ended up not doing it. Probably not within the scope of this PR but thought it worth sharing

**Comment by jgillies** [comment]
> @jgillies now that I'm thinking about this ... I'm not sure this is doing what you want. I think what you want is to have dbt-helper give you the _path_ to the file that you want to open so that you can open that in vim. Right now, I think vim is going to open a buffer and put the result of STDOUT (which is the _code_ for the model, but not the file where that code lives).
> 
> Does this make sense? Or do you have a different use-case in mind that i'm not understanding?

@mikekaminsky Yeah, what you're saying definitely makes sense, and that approach actually crossed my mind after I'd start futzing around with this. Opening the file directly would probably make more sense, and be easier to implement in Vim. But, I _think_ the results should be the same either way. Then again, returning the path might be a pretty niche use-case (not to say that printing the STDOUT isn't as well though...). So, that being said, I'd be happy with whichever approach is more aligned with how you envision dbt-helper working.

**Comment by mikekaminsky** [comment]
Okay I tried to re-work this a bit. I don't understand git so I ended up opening a separate PR (sorry!) here:
#20 

The wrinkle is that I'm not sure that the `--source` flag is working correctly (either right now or in that implementation). I had to comment out that line in the test [suite here](https://github.com/mikekaminsky/dbt-helper/pull/21/files#diff-597a83f4a2d118faf93cea66173f7c90R52)

@clrcrl knows the inner workings of this a bit better than I do, but I don't see an obvious way to get the correct source file for the package out of dbt. The path that is available in the `node` object is ` dbt_modules/local_dep/` which is not actually the source code for that model (it should be in `test/integration/007_open_test/local_dependency/models/my_package_model.sql`)

**Comment by mikekaminsky** [comment]
Closing this in favor of https://github.com/mikekaminsky/dbt-helper/pull/23

**Pull request closed without merging**